File: problems/cses/01-introductory_problems/1069-repetitions/main.py
import sys

CHUNK_SIZE = 3636


class RepetitionState:
    def __init__(self) -> None:
        self.max_streak = 0
        self.current_streak = 0
        self.last_char = None

    def consume(self, chunk: bytes) -> None:
        curr = self.current_streak
        last = self.last_char
        m_streak = self.max_streak

        for b in chunk:
            if b == last:
                curr += 1
            else:
                curr = 1
                last = b

            if curr > m_streak:
                m_streak = curr

        self.current_streak = curr
        self.max_streak = m_streak
        self.last_char = last


if __name__ == "__main__":
    state = RepetitionState()

    while True:
        chunk = sys.stdin.buffer.read(CHUNK_SIZE)
        if not chunk:
            break
        state.consume(chunk)

    print(state.max_streak)

File: problems/cses/01-introductory_problems/1069-repetitions/main.cpp
#include <iostream>
#include <string> 
#include <string_view>

struct RepetitionState {
    char last_char = '\0';
    size_t current_streak = 0;
    size_t max_streak = 0;

    void consume(std::string_view chunk) {
        for (char c: chunk) {
            if (c == last_char) 
                current_streak++;    
            else {
                current_streak = 1;
                last_char = c;
            }
            max_streak = std::max(max_streak, current_streak);
        }
    }
};


int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(NULL);

    RepetitionState state;

    char buffer[3636];
    while (std::cin.read(buffer, sizeof(buffer)) || std::cin.gcount() > 0) {
        state.consume(std::string_view(buffer, std::cin.gcount()));
    }

    std::cout << state.max_streak << '\n';
}

// https://cses.fi/problemset/task/1069/
